extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -0,0 +1,204 @@
+                        { error: err });
+                        return done(err);
+                    }
+                    this.logger.info('successfully deleted object from mongo',
+                    { key });
+                    return done();
+                });
+        }
+        if (sourceEntry instanceof ObjectQueueEntry) {
+            const bucket = sourceEntry.getBucket();
+            // always use versioned key so putting full version state to mongo
+            const key = sourceEntry.getObjectVersionedKey();
+            const objVal = sourceEntry.getValue();
+
+            // Always call putObject with version params undefined so
+            // that mongoClient will use putObjectNoVer which just puts
+            // the object without further manipulation/actions.
+            // S3 takes care of the versioning logic so consuming the queue
+            // is sufficient to replay the version logic in the consumer.
+            return this._mongoClient.putObject(bucket, key, objVal, undefined,
+                this.logger, err => {
+                    if (err) {
+                        this.logger.error('error putting object to mongo',
+                        { error: err });
+                        return done(err);
+                    }
+                    this.logger.info('successfully put object to mongo',
+                    { key });
+                    return done();
+                });
+        }
+        if (sourceEntry instanceof BucketMdQueueEntry) {
+            const masterBucket = sourceEntry.getMasterBucket();
+            const instanceBucket = sourceEntry.getInstanceBucket();
+            const val = sourceEntry.getValue();
+            return this._mongoClient.putObject(masterBucket,
+                instanceBucket, val, undefined,
+                this.logger, err => {
+                    if (err) {
+                        this.logger.error('error putting bucket ' +
+                        'metadata to mongo',
+                        { error: err, masterBucket, instanceBucket });
+                        return done(err);
+                    }
+                    this.logger.info('successfully put bucket' +
+                    ' metadata to mongo', { masterBucket, instanceBucket });
+                    return done();
+                });
+        }
+        if (sourceEntry instanceof BucketQueueEntry) {
+            const bucketOwnerKey = sourceEntry.getBucketOwnerKey();
+            const val = sourceEntry.getValue();
+            return this._mongoClient.putObject(usersBucket,
+                bucketOwnerKey, val, undefined,
+                this.logger, err => {
+                    if (err) {
+                        this.logger.error('error putting to usersbucket',
+                        { error: err, bucketOwnerKey });
+                        return done(err);
+                    }
+                    this.logger.info('successfully put to usersBucket',
```

Comment:
`usersBucket` is internal and should not be exposed in an info log. What about `bucket entry put into mongo`? Not convinced `entry` is the best wording, let you decide :)